de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/budgetissue/BudgetIssueRepository.java

```diff
@@ -0,0 +1,144 @@
+
+	private final ExternalIssueDetailsRepository externalIssueDetailsRepository;
+
+	public BudgetIssueRepository(final ExternalIssueDetailsRepository externalIssueDetailsRepository)
+	{
+		this.externalIssueDetailsRepository = externalIssueDetailsRepository;
+	}
+
+	public BudgetIssue store(final BudgetIssue budgetIssue)
+	{
+		final I_S_Budget_Issue record;
+
+		if (budgetIssue.getBudgetIssueId() != null)
+		{
+			record = InterfaceWrapperHelper.load(budgetIssue.getBudgetIssueId(), I_S_Budget_Issue.class);
+		}
+		else
+		{
+			record = InterfaceWrapperHelper.newInstance(I_S_Budget_Issue.class);
+		}
+		record.setAD_Org_ID(budgetIssue.getOrgId().getRepoId());
+
+		record.setName(budgetIssue.getName());
+		record.setDescription(budgetIssue.getDescription());
+		record.setBudget_Issue_Type(budgetIssue.getType().getValue());
+		record.setProcessed(budgetIssue.isProcessed());
+
+		record.setS_Current_Milestone_ID(NumberUtils.asInt(budgetIssue.getMilestoneId(), -1));
+
+		record.setEstimatedEffort(budgetIssue.getEstimatedEffort());
+
+		record.setBudgetedEffort(budgetIssue.getBudgetedEffort());
+
+		record.setAD_User_ID(NumberUtils.asInt(budgetIssue.getAssigneeId(), -1));
+
+		record.setExternalId(budgetIssue.getExternalIssueId());
+		record.setExternalIssueNo(budgetIssue.getExternalIssueNo());
+		record.setIssueURL(budgetIssue.getExternalIssueURL());
+
+		InterfaceWrapperHelper.save(record);
+
+		final BudgetIssueId budgetIssueId = BudgetIssueId.ofRepoId(record.getS_Budget_Issue_ID());
+		budgetIssue.setBudgetIssueId(budgetIssueId);
+
+		persistIssueDetails(budgetIssue);
+
+		return budgetIssue;
+	}
+
+	public Optional<BudgetIssue> getEntityByExternalId(@NonNull final String externalIssueId)
+	{
+		 final I_S_Budget_Issue budget_issue = queryBL
+				.createQueryBuilder(I_S_Budget_Issue.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_S_Budget_Issue.COLUMNNAME_ExternalId, externalIssueId)
+				.create()
+				.firstOnly(I_S_Budget_Issue.class);
+
+		 return budget_issue != null ? Optional.of(buildBudgetIssueEntity(budget_issue)) : Optional.empty();
+	}
+
+	private BudgetIssue buildBudgetIssueEntity(final I_S_Budget_Issue record)
+	{
+
+		return BudgetIssue.builder()
+				.orgId(OrgId.ofRepoId(record.getAD_Org_ID()))
+				.budgetIssueId(BudgetIssueId.ofRepoId(record.getS_Budget_Issue_ID()))
+				.effortUomId(UomId.ofRepoId(record.getEffort_UOM_ID()))
+				.milestoneId(MilestoneId.ofRepoIdOrNull(record.getS_Current_Milestone_ID()))
+				.assigneeId(UserId.ofNullableRepoId(record.getAD_User_ID()))
+				.name(record.getName())
+				.description(record.getDescription())
+				.type(BudgetIssueType.getTypeByValue(record.getBudget_Issue_Type()))
+				.processed(record.isProcessed())
+				.estimatedEffort(record.getEstimatedEffort())
+				.budgetedEffort(record.getBudgetedEffort())
+				.externalIssueId(record.getExternalId())
+				.externalIssueNo(record.getExternalIssueNo())
+				.externalIssueURL(record.getIssueURL())
```

Comment:
i think here you forgot to load the details

Reply:
Not really, there is just no need to load them at this point.

Reply:
ok, then pls rename the public method to soemthing like `getEntityWithoutDetailsByExternalId`

Reply:
@pvpurcarcosmin  @metas-jb  I disagree. 
That's a good sign that `externalIssueDetails` shall not be part of BudgetIssue class. The code is talking to u.

feel free to call me if u need details.